src/base/OpenMCProblemBase.C

```diff
@@ -116,11 +117,23 @@ OpenMCProblemBase::OpenMCProblemBase(const InputParameters & params)
     mooseError("The tally system used by OpenMCProblemBase derived classes has been deprecated. "
                "Please add tallies with the [Tallies] block instead.");
 
-  int argc = 1;
-  char openmc[] = "openmc";
-  char * argv[1] = {openmc};
+  std::vector<std::string> argv_vec = {"openmc"};
+
+  if (isParamSetByUser("xml_directory"))
```

Comment:
MOOSE convention is to fetch this parameter in the member initialization list instead - and, since you are setting a default, you actually don't need to check for the `isParamSetByUser` since presumably the parameter will always have a valid value unless the user decides to set it themselves.

Reply:
Thanks @TheBEllis! This looks great. Now that I think of it, I think it would be better for the test case to be entirely its own directory (since there's technically no way right now to detect if the case is actually using the XML files in the sub-folder if they are identical to the XML files in the working directory).

Would you please move the test case into its own folder, like under `neutronics`?

Also, I sent you an invite to join `cardinal-devs` - you'll need to accept this invite from github for the test suite to launch here.